var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -394,31 +406,31 @@ def edit(self, spec, prefix):
 
         if '+pexsi' in spec:
             cppflags.append('-D__LIBPEXSI')
-            fcflags.append('-I' + os.path.join(
+            fcflags.append('-I' + join_path(
                 spec['pexsi'].prefix, 'fortran'))
             libs.extend([
-                os.path.join(spec['pexsi'].libs.directories[0],
-                             'libpexsi.a'),
-                os.path.join(spec['superlu-dist'].libs.directories[0],
-                             'libsuperlu_dist.a'),
-                os.path.join(
+                join_path(spec['pexsi'].libs.directories[0], 'libpexsi.a'),
+                join_path(spec['superlu-dist'].libs.directories[0],
+                          'libsuperlu_dist.a'),
+                join_path(
                     spec['parmetis'].libs.directories[0],
                     'libparmetis.{0}'.format(dso_suffix)
                 ),
-                os.path.join(
+                join_path(
                     spec['metis'].libs.directories[0],
                     'libmetis.{0}'.format(dso_suffix)
                 ),
             ])
 
         if '+elpa' in spec:
+            dso_suffix = 'a' if '%aocc' in spec else 'so'
```

Comment:
I think this would be wrong on `platform=darwin`. Possibly:
```suggestion
            dso_suffix = 'a' if '%aocc' in spec else dso_suffix
```
would restore the old behavior for other compilers. Looking at the homepage it seems that `cp2k` may be used on Mac, though @dev-zero can comment more about that if needed.

As a minor comment, having a variable named `dso_suffix` containing the suffix for a static library may be confusing. This is to say that, if it is difficult to change the name of the variable you can maybe add a comment on why having static libraries is requested by `%aocc`, similarly to what was done for the `-Hx, ...` option above.

Reply:
yes, CP2K is used on macOS, but I doubt someone is going to use AOCC there.

What bothers me is the redefinition of the `dso_suffix` itself: should we decide to rearrange the blocks of code (`if ... in spec:`) or need `dso_suffix` later on, we might suddenly link other stuff statically.
And also: why?

Reply:
Removed overwriting condition of `dso_suffix` and implemented it in separate `if .... in spec:`